### ISSUE TITLE
Add login options to mapper frontend, remove temp login from React frontend

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#1890]
**Is your feature request related to a problem? Please describe.**

- Login is seamless between the two frontends.
- We want to remove temp login option from the management frontend, enforcing OSM login.
- The mapper frontend needs a way to login, without having to log in via React one first.

**Describe the solution you'd like**

- Remove temp login from React frontend.
- Update hotosm/ui header to include login logic (for mapper frontend).
- Options should be OSM and temp cookie for mappers.
- The temp cookie should not work when going back to the React frontend! (this might be tricky, possibly requiring checking for a separate cookie name _temp and _osm?)

**Comment by spwoodcock** [comment]
> Adding login on the mapper frontend is completed.
> 
> Also, temp login has been removed from the react frontend, but if the user logins via temp login from the mapper frontend and then redirects to the react frontend, the user is also logged on to the react frontend. 
> 
> So @spwoodcock, as you mentioned changing the cookie name to distinguish between OSM and temp login, I think firstly works on the backend has to be done as the cookie is being set from API call right?
> 
> You can review and merge this PR. Then on another PR, I'll do the remaining stuff after we finalize the process of validating osm & temp login on react frontend.

Haha well timed!!

**Comment by NSUWAL123** [comment]
Yeah :laughing:

**Comment by spwoodcock** [comment]
I made a separate issue for the backend cookie part. Playwright tests are addressed in another PR. This issue is solved once the linked PR is merged.
